### DISCUSSION TITLE
Does it support websocket?

### DISCUSSION THREAD

**Discussion by jasworks** [any1/wayvnc#353] [General]
Does it support websocket so that things like NoVNC can work in a browser?

**Comment by rtrial** [comment]
I have gotten NoVNC to work by disabling enable_auto=true in the config file.  However, connecting with RealVNC and UltraVNC viewers return complaints about no encryption (see the attached).

![Screenshot 2025-01-17 111034](https://github.com/user-attachments/assets/cf5287e1-d46a-47d3-b8d2-378c243a0303)

![Screenshot 2025-01-17 111030](https://github.com/user-attachments/assets/54dedb57-8293-435f-a4fe-b82942143ccf)

Those viewers do not require ID/password authentication with enable_auto=false.  

Is there a way to have NoVNC work and still require ID/password authentication?

**Reply by rtrial** [reply]
RealVNC message attached.
![Screenshot 2025-01-17 110909](https://github.com/user-attachments/assets/100233b7-4aaa-43c8-93b5-4a8979992011)

**Reply by any1** [reply]
```
enable_auth=true
relax_encryption=true
```

**Comment by rtrial** [comment]
My WayVNC config is here:
_use_relative_paths=true
address=::
enable_auth=true
relax_encryption=true
enable_pam=true
private_key_file=tls_key.pem
certificate_file=tls_cert.pem
rsa_private_key_file=rsa_key.pem_

RealVNC and UltraVNC viewers both connect with ID/password verification.  NoVNC still gives a "Failed to connect to server" error message.

**Comment by hrideshmg** [comment]
If anyone stumbles upon this thread, wayvnc does support websockets in 2026, you have to launch it with the `-w` flag. Here is a working config:
```
address=0.0.0.0
enable_auth=true
port=5900
use_relative_paths=true
username=<user>
password=<pass>
relax_encryption=true
```
the important line is `relax_encryption=true` wayvnc docs state that this enables some security types like Diffie-Helman, considering that noVNC also supports the same i assume that it is what wayvnc is using with this config.
